File: backend/src/main/java/com/cinebh/app/service/MovieService.java
package com.cinebh.app.service;

import com.cinebh.app.dto.MovieDto;
import com.cinebh.app.dto.PageDto;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

public interface MovieService {

    PageDto<MovieDto> getCurrentlyShowingMovies(
            String title, UUID cityId, UUID venueId, UUID genreId,
            LocalDate date, LocalTime time, Pageable pageable);

    PageDto<MovieDto> getUpcomingMovies(
            String title, UUID cityId, UUID venueId, UUID genreId,
            LocalDate startDate, LocalDate endDate, Pageable pageable);
}

File: frontend/src/services/movieService.js
import { api } from "./api";

export async function getCurrentlyShowingMovies({ title, cityId, venueId, genreId, date, time, page, size }) {
    try {
        const response = await api.get("/movies/currently-showing", {
            params: { title, cityId, venueId, genreId, date, time, page, size }
        });
        return response.data;
    }
    catch (error) {
        console.log("Error fetching currently showing movies:", error);
        throw error;
    }
}

export async function getUpcomingMovies({ title, cityId, venueId, genreId, startDate, endDate, page, size }) {
    try {
        const response = await api.get("/movies/upcoming", {
            params: { title, cityId, venueId, genreId, startDate, endDate, page, size }
        });
        return response.data;
    }
    catch (error) {
        console.log("Error fetching upcoming movies:", error);
        throw error;
    }
}

export async function getMovieDetails(movieId) {
    try {
        const response = await api.get(`/movies/${movieId}`);
        return response.data;
    }
    catch (error) {
        console.log("Error fetching movie details:", error);
        throw error;
    }
}

export async function getArchivedMovies({ page, size }) {
    try {
        const response = await api.get("/movies/archived", {
            params: { page, size }
        });
        return response.data;
    } catch (error) {
        console.log("Error fetching archived movies:", error);
        throw error;
    }
}

export async function archiveMovies(movieIds) {
    try {
        const response = await api.post("/movies/archive", movieIds);
        return response.data;
    } catch (error) {
        console.log("Error archiving movies:", error);
        throw error;
    }
}

export async function moveToDrafts(movieIds) {
    try {
        const response = await api.post("/movies/move-to-drafts", movieIds);
        return response.data;
    } catch (error) {
        console.log("Error moving movies to drafts:", error);
        throw error;
    }
}